optuna/trial/_fixed.py

```diff
@@ -79,52 +83,52 @@ def suggest_float(
 
         if step is not None:
             if log:
-                raise NotImplementedError(
-                    "The parameter `step` is not supported when `log` is True."
-                )
+                raise ValueError("The parameter `step` is not supported when `log` is True.")
```

Comment:
Might want to follow up this PR with tests for this logic since it seems like it's missing.